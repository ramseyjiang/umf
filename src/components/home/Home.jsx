import React from "react";
import { Container, ListGroup } from "react-bootstrap";

const Home = () => {
	const contents = {
		0: 'Allow User to Sign Up —> Any one can sign up',
		1: 'Allow Signed Up Users to be able to login. ',
		2: 'Allow Logged in users to log out.',
		3: 'List Users —> Any user (both Admin and normal user) can see list of users (provided they are logged in).',
		4: "Update a user —> Only an Admin User can Update any user's details (provided the Admin is logged in).",
		5: 'Delete a user —> Only an Admin User can Delete any user (provided the Admin is logged in).',
		6: 'Fields you should have against the User table are : id, email, username, firstName, lastName  password, created_at, updated_at, is_admin —> Required Fields',
		7: 'The fields created_at, updated_at should record Date and time as to when the record was created and updated.',
		8: 'Rebuild the Laravel auth default, make a user can login using username and password.',
		9: 'Deploy codes on AWS for preview and upload codes to GitHub for review.',
	}

	const colors = {
		0: 'success', 1:'danger', 2: 'warning', 3:'info', 4:'dark'
	}
  
  return (
    <>
      <Container className="home rounded">
				<h3 className="text-center">User Stories</h3>
        <div className="row">
          <div className="col-12">
            <ListGroup>
						{Object.keys(contents).map((key, index) => (
							<ListGroup.Item key={index} variant={colors[key%5]}>{contents[key]}</ListGroup.Item>
						))}
            </ListGroup>
          </div>
        </div>
      </Container>
    </>
  );
}
  
export default Home;