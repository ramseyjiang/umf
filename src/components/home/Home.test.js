import React from "react";
import renderer from "react-test-renderer";
import Home from "./Home";
import { render, screen } from "@testing-library/react";

describe("Home component test", () => {
  it("render correctly Home component", () => {
    const HomeComponent = renderer.create(<Home />).toJSON();
    expect(HomeComponent).toMatchSnapshot();
  });

  it("Home displays correct", () => {
    render(<Home />);
    expect(screen.getByText(/Allow User to Sign Up/i)).toBeInTheDocument();
    expect(screen.getByText(/Allow Signed Up Users to be able to login/i)).toBeInTheDocument();
    expect(screen.getByText(/Allow Logged in users to log out./i)).toBeInTheDocument();
    expect(screen.getByText(/List Users/i)).toBeInTheDocument();
    expect(screen.getByText(/Update a user/i)).toBeInTheDocument();
    expect(screen.getByText(/Delete a user/i)).toBeInTheDocument();
    expect(screen.getByText(/Fields you should have against the User table are/i)).toBeInTheDocument();
    expect(screen.getByText(/The fields created_at, updated_at should record Date and time /i)).toBeInTheDocument();
    expect(screen.getByText(/Rebuild the Laravel auth default/i)).toBeInTheDocument();
    expect(screen.getByText(/Deploy codes on AWS for preview and upload codes to GitHub for review./i)).toBeInTheDocument();
  });
});
