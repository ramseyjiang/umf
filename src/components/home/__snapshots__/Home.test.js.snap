// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component test render correctly Home component 1`] = `
<div
  className="home rounded container"
>
  <h3
    className="text-center"
  >
    User Stories
  </h3>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item list-group-item-success"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Allow User to Sign Up —&gt; Any one can sign up
        </div>
        <div
          className="list-group-item list-group-item-danger"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Allow Signed Up Users to be able to login. 
        </div>
        <div
          className="list-group-item list-group-item-warning"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Allow Logged in users to log out.
        </div>
        <div
          className="list-group-item list-group-item-info"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          List Users —&gt; Any user (both Admin and normal user) can see list of users (provided they are logged in).
        </div>
        <div
          className="list-group-item list-group-item-dark"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Update a user —&gt; Only an Admin User can Update any user's details (provided the Admin is logged in).
        </div>
        <div
          className="list-group-item list-group-item-success"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Delete a user —&gt; Only an Admin User can Delete any user (provided the Admin is logged in).
        </div>
        <div
          className="list-group-item list-group-item-danger"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Fields you should have against the User table are : id, email, username, firstName, lastName  password, created_at, updated_at, is_admin —&gt; Required Fields
        </div>
        <div
          className="list-group-item list-group-item-warning"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          The fields created_at, updated_at should record Date and time as to when the record was created and updated.
        </div>
        <div
          className="list-group-item list-group-item-info"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Rebuild the Laravel auth default, make a user can login using username and password.
        </div>
        <div
          className="list-group-item list-group-item-dark"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Deploy codes on AWS for preview and upload codes to GitHub for review.
        </div>
      </div>
    </div>
  </div>
</div>
`;
